src/SmartComponents/modules/reducers.js

```diff
@@ -1,14 +1,65 @@
 import types from './types';
 
 const initialState = {
-    compare: {},
+    fullCompareData: {},
     addSystemModalOpened: false,
     selectedSystemIds: [],
     filterDropdownOpened: false,
     stateFilter: 'all',
-    factFilter: ''
+    factFilter: '',
+    totalFacts: 0,
+    page: 1,
+    perPage: 10,
+    filteredCompareData: {}
 };
 
+function paginateFilteredData(data, selectedPage, factsPerPage) {
+    let paginatedData = {};
+    let paginatedFacts = [];
+    let divisor = 0;
+
+    for (let i = 0; i < data.facts.length; i++) {
+        divisor++;
+        if (Math.ceil(divisor / factsPerPage) === selectedPage) {
+            paginatedFacts.push(data.facts[i]);
+        }
+    }
+
+    paginatedData = {
+        facts: paginatedFacts,
+        systems: data.systems
+    };
+
+    return paginatedData;
+}
+
+function filterCompareData(data, stateFilter, factFilter) {
```

Comment:
If we just pass in `facts` here then we simplify this function and its scope of knowledge. That is, it only has to care about a facts data structure not a hash that has a `facts` key.

Reply:
Correct me if I'm wrong, but since we populate the DriftTable columns using the data.systems (https://github.com/RedHatInsights/drift-frontend/blob/master/src/SmartComponents/DriftPage/DriftTable/DriftTable.js#L51), then we'd want to build that filteredData with the facts and systems. If we only pass in facts, then we have to find a more convoluted way to build the filtered data outside of the function.

Reply:
Correct, but this function is *only* concerned with taking and filtering the set of facts into a subset of facts. Note that it only operates on `data.facts` so the `data.` part can be factored out.